auth/grpctransport/grpctransport.go

```diff
@@ -337,17 +337,19 @@ func (c *grpcCredentialsProvider) getClientUniverseDomain() string {
 }
 
 func (c *grpcCredentialsProvider) GetRequestMetadata(ctx context.Context, uri ...string) (map[string]string, error) {
-	credentialsUniverseDomain, err := c.creds.UniverseDomain(ctx)
-	if err != nil {
-		return nil, err
-	}
-	if err := transport.ValidateUniverseDomain(c.getClientUniverseDomain(), credentialsUniverseDomain); err != nil {
-		return nil, err
-	}
 	token, err := c.creds.Token(ctx)
 	if err != nil {
 		return nil, err
 	}
+	if source, _ := token.Metadata["auth.google.tokenSource"].(string); source != "compute-metadata" {
```

Comment:
These maps could be nil. Can you add a check.

Reply:
1. Would it be better to always initialize with an empty map?
2. Should I add the check to this existing usage as well? https://github.com/googleapis/google-cloud-go/blob/main/auth/grpctransport/directpath.go#L69-L74

Reply:
Users can create their own token types that we use, so best to be defensive.

Reply:
Updated, PTAL.